agent/tool-scripts/datalog/prometheus-metrics-datalog

```diff
@@ -15,39 +16,64 @@ key = sys.argv[7]
 if (cert == "None") or (key == "None"):
     command = "prom2json http://%s:%d/metrics" % (host, port)
 else:
-    command = "prom2json %s --cert=%s --key=%s https://%s:%d/metrics" % (
-            "--accept-invalid-cert" if selfsigned == "true" else "", cert, key, host, port)
+    command = "prom2json%s --cert=%s --key=%s https://%s:%d/metrics" % (
+            " --accept-invalid-cert" if selfsigned == "true" else "", cert,
+            key, host, port)
+
+_sleep_iterations = 0
```

Comment:
Can we squirrel this section away in a function to indicate that we are mocking `sleep` and `time`? Or at least add a comment to that effect?

Reply:
I added various comments to describe what was happening.  Moving this to a function or a class seemed like syntactic sugar given it is only used in this one place.